### PR TITLE
Updated readme to inform that clang 3.4 is minimum required version.

### PR DESCRIPTION
**Pull request by taylor24**

### CODE REVIEW

**Comment by Schwertspize** [comment]
If that is the appropriate location, we should also specify the minimum gcc version (r visual Studio)

**Comment by Seadragon91** [comment]
I think this should be specified in file `COMPILING.md`

**Comment by SphinxC0re** [comment]
@Seadragon91 I agree, this should definitively go into COMPILING.md

**Comment by taylor24** [comment]
Bear with me, this is my first pull request. What is the minimum version of gcc that should be required?

**Comment by Seadragon91** [comment]
We need gcc/g++ 4.8.

**Comment by LogicParrot** [comment]
@taylor24 Scroll down compiling.MD, you'll see this paragraph:

>(Ensure you have a recent version of clang, or gcc/g++ 4.8 or higher installed to avoid errors during compilation)

This  should be updated, and clang should be added. Also, the way you added the bullet point implies that clang is a *must*, instead, I'd modify the previous point e.g. like so:

>modern C++ compiler and linker (Clang 3.4+, GCC 4.8+, or VS 2013+).

**Comment by LogicParrot** [comment]
Also, do we really need a "C compiler"?

**Comment by Seadragon91** [comment]
We are using libraries, like lua that are using C.

**Comment by worktycho** [comment]
Merge.

File: COMPILING.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ Install git, make, cmake and clang (or gcc), using your platform's package manag
 ```
 sudo apt-get install git make cmake clang
 ```
-(Ensure you have a recent version of clang, or gcc/g++ 4.8 or higher installed to avoid errors during compilation)
+modern C++ compiler and linker (Clang 3.4+, GCC 4.8+, or VS 2013+).
```
Comment (LogicParrot) [inline COMPILING.md]: The sentence is gramatically incomplete.
"Ensure that you have modern C++ compiler and linker (Clang 3.4+, GCC 4.8+, or VS 2013+)"

**Comment by taylor24** [comment]
Thanks for letting me contribute to your guys' project. I'm just going to watch and see how open source projects are built

**Comment by LogicParrot** [comment]
Thank you for your contribution. Could you please rebase and squash this so that we can merge?

The procedure is as follows:

```
git fetch upstream master
git checkout master
git rebase -i upstream/master
```
A screen like below will pop up:
```
pick ...
pick ...
pick ...
```

Replace all occurences of `pick` with `s`, except the first one, then save. (In nano, saving is CTRL+X, then y)

**Comment by tigerw** [comment]
No need, GitHub can do it now.

**Comment by LogicParrot** [comment]
Magic!

**Pull request merged**